platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/TestEventReporterInternal.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing;
+
+import org.gradle.api.tasks.testing.TestEventReporter;
+import org.gradle.api.tasks.testing.TestFailure;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+import java.time.Instant;
+import java.util.List;
+
+@NullMarked
+public interface TestEventReporterInternal extends TestEventReporter {
```

Comment:
It seems the methods here can be moved to the parent interface. TestFailure is part of the public API and FWIU it can be useful for implementing 3rd party test integration.

Reply:
Sterling and I had concerns over directly exposing `TestFailure` in this interface since it's supposed to be generic and not JVM-specific, but `TestFailure` requires a `Throwable`. Maybe at this point it would be worth just exposing it though, and we can refined `TestFailure` to make it non-JVM-specific another time.